src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTask.java

```diff
@@ -0,0 +1,421 @@
+
+        Stopwatch bulkDownloadStopwatch = Stopwatch.createStarted();
+        BulkFileDownloadResponse bulkDownloadResponse;
+        try {
+            bulkDownloadResponse = synapseHelper.generateBulkDownloadFileHandle(params.getSynapseTableId(),
+                    ctx.getFileHandleIdSet());
+            ctx.setFileSummaryList(bulkDownloadResponse.getFileSummary());
+
+            String bulkDownloadFileHandleId = bulkDownloadResponse.getResultZipFileHandleId();
+            synapseHelper.downloadFileHandle(bulkDownloadFileHandleId, bulkDownloadFile);
+            ctx.setBulkDownloadFile(bulkDownloadFile);
+        } catch (AsyncTimeoutException | SynapseException ex) {
+            throw new AsyncTaskExecutionException("Error bulk downloading file handles to file " +
+                    bulkDownloadFilePath + ": " + ex.getMessage(), ex);
+        } finally {
+            bulkDownloadStopwatch.stop();
+            LOG.info("Bulk downloading file handles to file " + bulkDownloadFilePath + " took " +
+                    bulkDownloadStopwatch.elapsed(TimeUnit.MILLISECONDS) + " ms");
+        }
+    }
+
+    /**
+     * <p>
+     * We need to make edits to the CSV: (1) Replace the file handle IDs with zip entry names. (2) Remove health
+     * codes, since those aren't supposed to be exposed to users. This method reads from
+     * {@link SynapseDownloadFromTableParameters#getTempDir} to determine where to write the edited CSV and from
+     * {@link SynapseDownloadFromTableParameters#getSchema} to generate the edited CSV file name. It also reads from
+     * {@link SynapseDownloadFromTableContext#getFileSummaryList},
+     * {@link SynapseDownloadFromTableContext#getColumnInfo}, {@link SynapseDownloadFromTableContext#getCsvFile}, and
+     * writes the results to {@link SynapseDownloadFromTableContext#setEditedCsvFile}.
+     * </p>
+     * <p>
+     * This method is package-scoped, to allow unit tests to inject an exception here.
+     * </p>
+     */
+    void editCsv() {
+        // Convert file summary in bulk download response into a map from file handle ID to zip entry name.
+        Map<String, String> fileHandleIdToReplacement = new HashMap<>();
+        List<FileDownloadSummary> fileSummaryList = ctx.getFileSummaryList();
+        if (fileSummaryList != null) {
+            for (FileDownloadSummary oneFileSummary : fileSummaryList) {
+                String fileHandleId = oneFileSummary.getFileHandleId();
+                if (!Strings.isNullOrEmpty(fileHandleId)) {
+                    String zipEntryName = oneFileSummary.getZipEntryName();
+                    String failureMessage = oneFileSummary.getFailureMessage();
+
+                    if (!Strings.isNullOrEmpty(zipEntryName)) {
+                        // replace file handle ID with zip entry name
+                        fileHandleIdToReplacement.put(fileHandleId, zipEntryName);
+                    } else if (!Strings.isNullOrEmpty(failureMessage)) {
```

Comment:
The only user-entered data should be the startDate and endDate in the BridgeUddRequest, which get converted to LocalDates immediately. So it shouldn't matter. I just used Strings.isNullOrEmpty() everywhere to catch more error cases.

One could argue we don't need this at all and could just do straight null checks. One could also argue that we should go all the way up and use Apache StringUtils.

Based on our earlier conversation, it sounds like this is relatively minor, and that it's prolly not worth going back and re-doing all of these.